tests/platform_tests/api/test_thermal.py

```diff
@@ -45,6 +53,27 @@ def setup(self, platform_api_conn):
             except:
                 pytest.fail("num_thermals is not an integer")
 
+    #
+    # Helper functions
+    #
+
+    def compare_value_with_platform_facts(self, key, value, thermal_idx):
+        expected_value = None
+
+        if self.chassis_facts:
+            expected_thermals = self.chassis_facts.get("thermals")
+            if expected_thermals:
+                expected_value = expected_thermals[thermal_idx].get(key)
+
+
+        if not expected_value:
+            logger.warning("Unable to get expected value for '{}' from platform.json file for thermal {}".format(key, thermal_idx))
+            return
```

Comment:
Remove warning message and make the check mandatory. See SFP portion of https://github.com/Azure/sonic-mgmt/pull/2268 for examples.

Reply:
fixed